src/Microsoft.DotNet.Cli.Utils/CommandResolution/ProjectToolsCommandResolver.cs

```diff
@@ -323,11 +330,61 @@ private IEnumerable<string> GetPossiblePackageRoots(IProject project)
                 .Build(toolLibrary, null, toolLockFile, framework, null);
 
             var tempDepsFile = Path.GetTempFileName();
-            using (var fileStream = File.Open(tempDepsFile, FileMode.Open, FileAccess.Write))
+
+            var args = new List<string>();
+
+            args.Add(toolDepsJsonGeneratorProject);
+            args.Add($"/p:ProjectAssetsFile=\"{toolLockFile.Path}\"");
+            args.Add($"/p:ToolName={toolLibrary.Name}");
+            args.Add($"/p:ProjectDepsFilePath={tempDepsFile}");
+
+            var toolTargetFramework = toolLockFile.Targets.First().TargetFramework.GetShortFolderName();
+            args.Add($"/p:TargetFramework={toolTargetFramework}");
+
+
+            //  Look for the .props file in the Microsoft.NETCore.App package, until NuGet
+            //  generates .props and .targets files for tool restores (https://github.com/NuGet/Home/issues/5037)
+            var platformLibrary = toolLockFile.Targets
+                .FirstOrDefault(t => framework == t.TargetFramework)
+                ?.GetPlatformLibrary();
+
+            if (platformLibrary != null)
             {
-                var dependencyContextWriter = new DependencyContextWriter();
+                string buildRelativePath = platformLibrary.Build.FirstOrDefault()?.Path;
 
-                dependencyContextWriter.Write(dependencyContext, fileStream);
+                var platformLibraryPath = toolLockFile.GetPackageDirectory(platformLibrary);
+
+                if (platformLibraryPath != null && buildRelativePath != null)
+                {
+                    //  Get rid of "_._" filename
+                    buildRelativePath = Path.GetDirectoryName(buildRelativePath);
+
+                    string platformLibraryBuildFolderPath = Path.Combine(platformLibraryPath, buildRelativePath);
+                    var platformLibraryPropsFile = Directory.GetFiles(platformLibraryBuildFolderPath, "*.props").FirstOrDefault();
```

Comment:
Depends on how temporary this is.  You should at least be able to use the package ID instead of `*` that's a safe assumption to make since it's a nuget requirement.

A slightly better pattern would be the following: `<packageFolder>\build\<tfm>\<packageId>.props`

A even better algo would be to use [FindBestItemGroup](https://github.com/NuGet/NuGet.Client/blob/1a1c474aab5c1e851b416c583ca4c6d727bd94e3/src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs#L85) passing in [ManagedCodeConventions.Patterns.MSBuildFiles](https://github.com/NuGet/NuGet.Client/blob/1a1c474aab5c1e851b416c583ca4c6d727bd94e3/src/NuGet.Core/NuGet.Packaging/ContentModel/ManagedCodeConventions.cs#L373), but not so sure you want to take a NuGet dependency here.